src/tree_model_sort.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2013-2015, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib::object::{Downcast, IsA};
+use glib::translate::*;
+
+use {TreeModel, TreeModelSort};
+
+impl TreeModelSort {
+    pub fn new<T: IsA<TreeModel>>(child_model: &T) -> TreeModelSort {
+        skip_assert_initialized!();
+        unsafe {
+            TreeModel::from_glib_none(ffi::gtk_tree_model_sort_new_with_model(child_model.to_glib_none().0))
```

Comment:
Wo, this code formatting looks super weird. Please change it to:

```rust
impl TreeModelSort {
    pub fn new<T: IsA<TreeModel>>(child_model: &T) -> TreeModelSort {
        skip_assert_initialized!();
        unsafe {
            TreeModel::from_glib_none(
                ffi::gtk_tree_model_sort_new_with_model(
                    child_model.to_glib_none().0)
                ).downcast_unchecked()
        }
    }
}
```

Reply:
I'll change this. FYI I mostly copied that from `TreeModelFilter`.

Reply:
If the code looks like this, I'm ashamed we accepted it. :O